### DISCUSSION TITLE
[react-table bug] expand and collapse all rows doesn't always work first time

### DISCUSSION THREAD

**Discussion by d-mather** [TanStack/table#3637] [General]
Hey there,

I noticed this first with my own code, then tried it out in the official example [here](https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/expanding), and was wondering if it is a known issue or currently being fixed or if I'm doing something wrong on my side.

**Here is what happens and how to reproduce it:**
When all the react-table rows are collapsed, manually go through all rows/sub rows and expand everything individually. Once that is done, notice that (if present) the expand/collapse all button did not toggle to reflect isAllRowsExpanded == true. If you click this now, then it quickly jumps to the right state (isAllRowsExpanded == true) but the rows stay the same. You will have to click on it a second time in order for all the rows to be collapsed again, making isAllRowsExpanded == false.

_Please could anyone advise on this issue._
I see it's been raised before [here](https://github.com/tannerlinsley/react-table/issues/3061) and [here](https://github.com/tannerlinsley/react-table/issues/3008), but still no outcome :(

Kind Regards
Dillon

**Comment by stuthib** [comment]
@d-mather I am running into this issue as well. I have deeply nested rows. It only works if I have already clicked on each row. On the initial load, expand all is not showing the rows at a depth of two or more. Are you still running into this issue? Not sure how to proceed.
